File: java/IncreasingArray.java
import java.util.Scanner;

public class IncreasingArray {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] arr = new int[n];

        for (int i = 0; i < n; i++) {
            arr[i] = scanner.nextInt();
        }

        long ans = 0;
        for (int i = 1; i < n; i++) {
            if (arr[i] < arr[i - 1]) {
                ans += arr[i - 1] - arr[i];
                arr[i] = arr[i - 1];
            }
        }

        System.out.println(ans);
    }
}

File: cpp/Increasing Array.cpp

#include<bits/stdc++.h>
using namespace std;

#define  ll long long


int main(){
	int n; cin >> n;
	int arr[n];
	for (int i = 0; i < n; i ++){
		cin >> arr[i];
	}

	ll ans = 0;
	for (int i = 1; i < n; i ++){
		if (arr[i - 1] > arr[i]){
			ans += (ll)(arr[i - 1] - arr[i]);
			arr[i] = arr[i - 1];
		}
	}
	cout << ans << "\n";
}